server/api/jellyfin.ts

```diff
@@ -0,0 +1,246 @@
+  agent: string;
+}
+
+export interface JellyfinLibraryItem {
+  Name: string;
+  Id: string;
+  HasSubtitles: boolean;
+  Type: 'Movie' | 'Episode' | 'Season' | 'Series';
+  SeriesName?: string;
+  SeriesId?: string;
+  SeasonId?: string;
+  SeasonName?: string;
+  IndexNumber?: number;
+  ParentIndexNumber?: number;
+  MediaType: string;
+}
+
+export interface JellyfinMediaStream {
+  Codec: string;
+  Type: 'Video' | 'Audio' | 'Subtitle';
+  Height?: number;
+  Width?: number;
+  AverageFrameRate?: number;
+  RealFrameRate?: number;
+  Language?: string;
+  DisplayTitle: string;
+}
+export interface JellyfinMediaSource {
+  Protocol: string;
+  Id: string;
+  Path: string;
+  Type: string;
+  VideoType: string;
+  MediaStreams: JellyfinMediaStream[];
+}
+
+export interface JellyfinLibraryItemExtended extends JellyfinLibraryItem {
+  ProviderIds: {
+    Tmdb?: string;
+    Imdb?: string;
+    Tvdb?: string;
+  };
+  MediaSources?: JellyfinMediaSource[];
+  Width?: number;
+  Height?: number;
+  IsHD?: boolean;
+  DateCreated?: string;
+}
+class JellyfinAPI {
+  private authToken?: string;
+  private jellyfinHost: string;
+  private axios: AxiosInstance;
+
+  constructor(jellyfinHost: string, authToken?: string) {
+    this.jellyfinHost = jellyfinHost;
+    this.authToken = authToken;
+
+    let authHeaderVal = '';
+    if (this.authToken) {
+      authHeaderVal =
+        'MediaBrowser Client="Jellyfin Web", Device="Firefox", DeviceId="TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6ODUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC84NS4wfDE2MTI5MjcyMDM5NzM1", Version="10.8.0", Token="' +
+        authToken +
+        '"';
+    } else {
+      authHeaderVal =
+        'MediaBrowser Client="Jellyfin Web", Device="Firefox", DeviceId="TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6ODUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC84NS4wfDE2MTI5MjcyMDM5NzM1", Version="10.8.0"';
+    }
+
+    this.axios = axios.create({
+      baseURL: this.jellyfinHost,
+      headers: {
+        'X-Emby-Authorization': authHeaderVal,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    });
+  }
+
+  public async login(
+    Username?: string,
+    Password?: string
+  ): Promise<JellyfinLoginResponse> {
+    try {
+      const account = await this.axios.post<JellyfinLoginResponse>(
+        '/Users/AuthenticateByName',
+        {
+          Username: Username,
+          Pw: Password,
+        }
+      );
+      return account.data;
+    } catch (e) {
+      throw new Error('Unauthorized');
+    }
+  }
+
+  public async getUser(): Promise<JellyfinUserResponse> {
+    try {
+      const account = await this.axios.get<JellyfinUserResponse>('/Users/Me');
+      return account.data;
+    } catch (e) {
+      logger.error(
+        `Something went wrong while getting the account from the Jellyfin server: ${e.message}`,
+        { label: 'Jellyfin API' }
+      );
+      throw new Error('Invalid auth token');
+    }
+  }
```

Comment:
This doesn't seem to be valid. We'd have to trust the token we get from `/Users/AuthenticateByName` in the same way that we do with Plex.

Reply:
what do you mean?

Reply:
resolved in Discord